src/Microsoft.AspNet.Server.Kestrel/Filter/StreamSocketOutput.cs

```diff
@@ -30,5 +33,27 @@ Task ISocketOutput.WriteAsync(ArraySegment<byte> buffer, bool immediate, Cancell
             _outputStream.Write(buffer.Array, buffer.Offset, buffer.Count);
             return TaskUtilities.CompletedTask;
         }
+
+        public MemoryPoolIterator2 ProducingStart()
+        {
+            _producingBlock = _memory.Lease();
+            return new MemoryPoolIterator2(_producingBlock);
+        }
+
+        public void ProducingComplete(MemoryPoolIterator2 end, int count)
```

Comment:
I think in practice this will only be used for what were `immediate: false` writes. We can certainly make an async version of this if we find the need.